picard/util/textencoding.py

```diff
@@ -181,19 +181,17 @@ def unicode_simplify_compatibility(string):
 }
 
 
-def _replace_unicode_simplify_punctuation(char, pathsave, win_compat):
-    result = _simplify_punctuation.get(char)
-    if result is None:
-        return char
-    elif not pathsave:
-        return result
-    else:
-        return sanitize_filename(result, win_compat=win_compat)
-
-
 def unicode_simplify_punctuation(string, pathsave=False, win_compat=False):
-    return ''.join(
-        _replace_unicode_simplify_punctuation(c, pathsave, win_compat) for c in string)
+    temp = []
+    for c in string:
+        try:
+            result = _simplify_punctuation[c]
+        except KeyError:
+            result = c
+        if c != result and pathsave:
```

Comment:
If `KeyError` was raised, we are 100% `c == result` so I think you can move this in the try part.

```python
        try:
            result = _simplify_punctuation[c]
            if c != result and pathsave:
                result = sanitize_filename(result, win_compat=win_compat)
        except KeyError:
            result = c
```

Reply:
OK